src/sidebar/helpers/export-annotations.ts

```diff
@@ -0,0 +1,49 @@
+import type { Annotation } from '../../types/api';
+
+export type ExportAnnotationsOptions = {
+  annotations: Annotation[];
+  clientVersion: string;
+  userId: string;
+};
+
+function buildExportFileContents(
+  { annotations, clientVersion, userId }: ExportAnnotationsOptions,
+  now: Date
+): string {
+  const data = {
+    export_date: now.toISOString(),
+    export_userid: userId,
+    client_version: clientVersion,
+    annotations,
+  };
+
+  return JSON.stringify(data, null, 2);
+}
+
+/**
+ * @param _document - Test seam
+ * @param now - Test seam
+ */
+export function exportAnnotations(
+  options: ExportAnnotationsOptions,
+  _document = document,
+  now = new Date()
+): string {
+  const link = _document.createElement('a');
+  const fileContent = buildExportFileContents(options, now);
+  const blob = new Blob([fileContent], {
+    type: 'application/json',
+  });
+  const url = URL.createObjectURL(blob);
+  const filename = 'annotations-export.json'; // TODO To be defined
+
+  link.setAttribute('href', url);
+  link.setAttribute('download', filename);
+  link.style.visibility = 'hidden';
+
+  _document.body.appendChild(link);
+  link.click();
+  _document.body.removeChild(link);
+
+  return fileContent;
+}
```

Comment:
This function takes a slightly different approach to what was prototyped here https://github.com/hypothesis/client/compare/main...spike-data-download

The reason is that with this approach, the exportAnnotations function can be called from anywhere, and it's not bound to a specific existing DOM element. Instead it creates a dummy link that can be used to trigger the download.

The mechanism to make the link "invisible" but actionable is a bit naive (hidden visibility), so we may need to iterate over this.

The purpose of returning the serialized content of the file is mostly to help testing it.